### PR TITLE
bash (and zsh) completion

### PR DESCRIPTION
**Pull request by rgbkrk**
This adds bash completion, straight from `nvm` but with some trimming. I've also fleshed out the README. We still have documented features that aren't implemented, so we'll need to trim those out later.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,78 @@
...
+
+Note: `dvm` does not support Windows or Fish shell.
+
+### Manual install
+
+If you have `git` installed, then clone this repository to `~/.dvm`.
+
+```
+git clone https://github.com/rgbkrk/dvm.git ~/.dvm
+# TODO: git checkout `git describe --abbrev=0 --tags`
```
Comment (smashwilson) [inline README.md]: Pssht. Master checkout all the way.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,78 @@
...
+
+Add this line to your `~/.bashrc`, `~/.profile`, or `~/.zshrc` file to have it automatically sourced upon login.
+
+## Usage
+
+To install the 1.8.2 release of docker, do this:
+
+    dvm install 1.8.2
+
+Now in any new shell use the installed version
```
Comment (smashwilson) [inline README.md]: Need a `:` here.

File: dvm.sh [hunk 3]
```diff
@@ -290,8 +290,8 @@ dvm() {
         echo "Example:"
         echo "  dvm install 1.8.1                     Install a specific version number"
         echo "  dvm use 1.6                           Use the latest available 1.6.x release"
-        echo "  dvm run 1.6.1 -it ubuntu /bin/bash    Run an ubuntu container using Docker 1.6.1"
-        echo "  dvm exec 1.6.1 nginx                  Run the nginx image the PATH pointing to Docker 1.6.1"
+        echo "  dvm run 1.6.1 run -it ubuntu bash     Run an ubuntu container using Docker 1.6.1"
+        echo "  dvm exec 1.6.1 run nginx              Run the nginx image the PATH pointing to Docker 1.6.1"
```
Comment (smashwilson) [inline dvm.sh]: Ooh cool. Good idea.

**Comment by rgbkrk** [comment]
Addressed comments.

**Comment by smashwilson** [comment]
:metal:

**Pull request merged**